model/value.go

```diff
@@ -221,80 +190,76 @@ func (s Samples) Equal(o Samples) bool {
 
 // SampleStream is a stream of Values belonging to an attached COWMetric.
 type SampleStream struct {
-	Metric Metric       `json:"metric"`
-	Values []SamplePair `json:"values"`
+	Metric     Metric                `json:"metric"`
+	Values     []SamplePair          `json:"values"`
+	Histograms []SampleHistogramPair `json:"histograms"`
 }
 
 func (ss SampleStream) String() string {
-	vals := make([]string, len(ss.Values))
+	valuesLength := len(ss.Values)
+	vals := make([]string, valuesLength+len(ss.Histograms))
 	for i, v := range ss.Values {
 		vals[i] = v.String()
 	}
+	for i, v := range ss.Histograms {
```

Comment:
I think we should have the samples in the order of timestamps. But grouping with type might be more helpful as well. @beorn7 do you have any opinion on this?

Reply:
The HTTP query API is first grouping by type, and then only within each type, it returns samples in the order of timestamps, https://prometheus.io/docs/prometheus/latest/querying/api/#range-vectors